### DISCUSSION TITLE
OSC out ops

### DISCUSSION THREAD

**Discussion by cristianvogel** [cables-gl/cables_docs#126] [Feature Requests]
I want to format data in a Cables patch as OSC  and send via WebSocket through the [osc4ws](https://github.com/cristianvogel/osc4ws) tool I wrote . But I feel I am jumping through hoops with the standard ops, perhaps I am not seeing the right way to do it?   Or perhaps we need some OSC out ops to plug into `WebSocketSend` op?

I'd like to send out  data like

```{ 
      "address" : "/anOscAddress",
      "args" : { {"type" : "f", "value" : 0.5} }
}
```
or

```{ 
      "address" : "/anOscAddress",
      "args" : { {"type" : "i", "value" : 5000}, {"type" : "f", "value" : 0.333} }
}
```

or

```{ 
      "address" : "/anOscAddress",
      "args" : { "type" : "s", "value" : "some string" }
}
```

and even blobs

```{ 
      "address" : "/anOscAddress",
      "args" : { "type" : "b", "value" : "arrayAsBlob" }
}
```

of course being able to inject changing values where 'value' keys are

**Comment by steam0r** [comment]
hey,

since all OSC is via websockets anyhow, you could use 

https://cables.gl/op/Ops.Net.WebSocket.WebSocketSend

right?

**Reply by cristianvogel** [reply]
Hi. Yes. Now you can with [osc4ws](https://github.com/cristianvogel/osc4ws) - before you couldn't. 
So , what I am asking is to make it easier to create OSC formatted data objects in Cables, before they get sent to the WebSocket. They need to be in a very precise format https://hangar.org/wp-content/uploads/2012/01/The-Open-Sound-Control-1.0-Specification-opensoundcontrol.org_.pdf    

Even just a simple OSC message with an address and an array of 1 value,  quickly gets hairy. 

Usually, you'd be wanting to wire up quite a few of them. 
<img width="774" alt="Screenshot 2022-01-23 at 22 50 38" src="https://user-images.githubusercontent.com/12829050/151243625-073edfea-e30f-41f2-987d-5aea2225bb77.png">

**Reply by steam0r** [reply]
see other discussion ;) #131
